fvid/fvid.py

```diff
@@ -128,19 +203,63 @@ def save_bits_to_file(file_path, bits):
     ) as f:
         bitstring.tofile(f)
 
+def save_bits_to_file_crypto(file_path, bits, key):
+    bitstring_temp = Bits(bin=bits)
+    encrypted = bitstring_temp.tobytes()
+
+    # #zip
+    # in_ = io.BytesIO()
+    # in_.write(encrypted)
+    # in_.seek(0)
+    # with gzip.GzipFile(fileobj=in_, mode='rb') as fo:
+        # encrypted = fo.read()
+    # #zip
+
+    f = Fernet(key)
+    decrypted_bits = f.decrypt(encrypted)#.decode()
+    _dict = pickle.loads(decrypted_bits)
+    filename = _dict['filename']
+    decrypted_bits_with_tail = _dict['data']
+    
+    bitstring_with_tail = Bits(bin=decrypted_bits_with_tail)
+    bitstring_with_tail = bitstring_with_tail.bin
+    # print('decoded_bitstring', bitstring_with_tail)
+    delimiter_str = DELIMITER.replace("0b", "")
+    delimiter_length = len(delimiter_str)
+
+    if bitstring_with_tail[-delimiter_length:] == delimiter_str:
```

Comment:
I don't think this will work, which is the same as what is on line **184**, but because of the error on **218**, I haven't tested it. 

The reason is that when **get_bits_from_video** is called with the data has been encrypted;  what ends up happening is that the delimiter is never found and therefore if you see the output to the fvid_frames directory, we might actually have duplicate frames. ultimately, when **save_bits_to_file_crypto** is called there is actually more data in there that the file originally had since all the frames (including any duplicate frames) will become part of the data.

For reference purposes, here is the file I used to test it: https://upload.wikimedia.org/wikipedia/en/7/7d/Lenna_%28test_image%29.png 

The program ran for me for 20 frames (as indicated by the number of loading bars when getting frames), but if you stop the program from decoding when that is happening you will find that in the fvid_frames directory there actually 12 frames that contain the data (indicated by that frame #12 has data only part of the frame and the rest is black):

![decoded_frames012](https://user-images.githubusercontent.com/18052272/95635905-993d9000-0a42-11eb-9bf7-4f8eaca2a13b.png)

So again, if we are getting the data for all 20 frames, then even if we try to find the delimiter, we might have extra data. 

but it's hard to say for sure without being able to decode the data from the issue I am getting. So if you fix that we should be able to see if I am correct or not.